ambry-rest/src/main/java/com.github.ambry.rest/NettyServerFactory.java

```diff
@@ -78,11 +80,39 @@ public NettyServerFactory(VerifiableProperties verifiableProperties, MetricRegis
   }
 
   /**
+   * Initialize the {@link ChannelHandler}s to be used in the netty pipeline
+   * @param nettyConfig the {@link NettyConfig} instance that defines the configuration parameters for the NettyServer.
+   * @param nettyMetrics the {@link NettyMetrics} instance to use to record metrics.
+   * @param requestHandler the {@link RestRequestHandler} that can be used to submit requests that need to be handled.
+   * @param publicAccessLogger the {@link PublicAccessLogger} that can be used for public access logging
+   * @param restServerState the {@link RestServerState} that can be used to check the health of the system
+   *                              to respond to health check requests
+   */
+  private void initializeChannelHandlers(NettyMetrics nettyMetrics, NettyConfig nettyConfig,
```

Comment:
Good point. We are providing two levels of flexibility. One at the Factory Level and one at the Server Level. Either users can define their own NioServer and NioServerFactory. Or they can use the readily available NettyServer with their own custom handlers. So, this patch is about the latter.